packages/flutter/test/widgets/grid_view_test.dart

```diff
@@ -899,4 +899,134 @@ void main() {
     expect(controller.position.maxScrollExtent, 472.0);
     expect(controller.position.pixels, 472.0);
   });
+
+  testWidgets(
+      'SliverGridDelegateWithFixedCrossAxisCount mainAxisExtent corrected',
+      (WidgetTester tester) async {
+    // Regression test for https://github.com/flutter/flutter/issues/138871
+    const SliverGridDelegateWithFixedCrossAxisCount delegate =
+        SliverGridDelegateWithFixedCrossAxisCount(
+      crossAxisCount: 3,
+      mainAxisSpacing: 8,
+      crossAxisSpacing: 8,
+      mainAxisExtent: -100,
```

Comment:
I think this should probably assert, this does not seem like a valid main axis extent.

Reply:
I added assertions and adjusted the test cases